src/app/border_agent.cpp

```diff
@@ -228,6 +228,21 @@ static int HandleRecord(const struct sockaddr *from,
             {
                 VerifyOrExit(value == "1", error = ERROR_BAD_FORMAT("value of TXT Key 'rv' is {} but not 1", value));
             }
+            else if (key == "dd")
+            {
+                auto &discriminator = binaryValue;
+
+                if (discriminator.size() != kExtendedAddrLength)
+                {
+                    ExitNow(error = ERROR_BAD_FORMAT("value of TXT Key 'dd' is invalid: value={}",
```

Comment:
```suggestion
                    borderAgent.mPresentFlags &= ~BorderAgent::kDiscriminatorBit;
                    ExitNow(error = ERROR_BAD_FORMAT("value of TXT Key 'dd' is invalid: value={}",
```

Necessary to mark it as not present?

Reply:
The `BorderAgent` default constructor initializes the flags to all zeros, so not necessary here.